language/move-lang/tests/functional/validator_set/register_validator.move

```diff
@@ -0,0 +1,80 @@
+//! account: alice
+//! account: vivian, 100000, 0, validator
+//! account: viola, 100000, 0, validator
+
+// check that the validator account config works
+script{
+    use 0x0::LibraSystem;
+
+    fun main() {
+        0x0::Transaction::assert(!LibraSystem::is_validator(0x0::Transaction::sender()), 1);
+        0x0::Transaction::assert(!LibraSystem::is_validator({{alice}}), 2);
+        0x0::Transaction::assert(LibraSystem::is_validator({{vivian}}), 3);
+        0x0::Transaction::assert(LibraSystem::is_validator({{viola}}), 4);
+        // number of validators should equal the number we declared
+        0x0::Transaction::assert(LibraSystem::validator_set_size() == 2, 5);
+        0x0::Transaction::assert(LibraSystem::get_ith_validator_address(1) == {{vivian}}, 6);
+        0x0::Transaction::assert(LibraSystem::get_ith_validator_address(0) == {{viola}}, 7);
+    }
+}
+
+// check: EXECUTED
+
+//! new-transaction
+//! sender: vivian
+script{
+    use 0x0::LibraSystem;
+
+    // check that sending from validator accounts works
+    fun main() {
+        0x0::Transaction::assert(LibraSystem::is_validator(0x0::Transaction::sender()), 8);
+    }
+}
+
+// check: EXECUTED
+
+//! new-transaction
+//! sender: association
+script{
+    use 0x0::LBR;
+    use 0x0::LibraAccount;
+
+    // register Alice as a validator candidate
+    fun main() {
+        LibraAccount::create_validator_account<LBR::T>(0xAA, x"00000000000000000000000000000000");
+    }
+}
+
+// check: EXECUTED
+
+// TODO(valerini): enable the following test once the sender format is supported
+// //! new-transaction
```

Comment:
same here, also why 0xaa versus a name?

Reply:
The name can't be used because the account for the name already exists... I can't create another account say for {{alice}} address... I need a fresh name to create an address, and fresh names aren't supported for these tests yet.